File: src/weaponsprocurement/gui/WimGuiToggleHeading.java
package weaponsprocurement.gui;

final class WimGuiToggleHeading {
    private WimGuiToggleHeading() {
    }

    static String label(String title, boolean expanded) {
        return safeTitle(title) + " " + suffix(expanded);
    }

    static String countedLabel(String title, int count, boolean expanded) {
        return safeTitle(title) + " [" + Math.max(0, count) + "] " + suffix(expanded);
    }

    private static String suffix(boolean expanded) {
        return expanded ? "(-)" : "(+)";
    }

    private static String safeTitle(String title) {
        return title == null ? "" : title;
    }
}

File: src/main/kotlin/weaponsprocurement/gui/WimGuiToggleHeading.kt
package weaponsprocurement.gui

class WimGuiToggleHeading private constructor() {
    companion object {
        @JvmStatic
        fun label(title: String?, expanded: Boolean): String = safeTitle(title) + " " + suffix(expanded)

        @JvmStatic
        fun countedLabel(title: String?, count: Int, expanded: Boolean): String {
            return safeTitle(title) + " [" + Math.max(0, count) + "] " + suffix(expanded)
        }

        private fun suffix(expanded: Boolean): String = if (expanded) "(-)" else "(+)"

        private fun safeTitle(title: String?): String = title ?: ""
    }
}
